### PR TITLE
Master stock split moves kic

### PR DESCRIPTION
**Pull request by kirti-c**
Description of the issue/feature this PR addresses:
Task ID - https://www.odoo.com/web#id=1912896&action=327&model=project.task&view_type=form&menu_id=4720
Pad - https://pad.odoo.com/p/r.83e35586ce65fb5c181f217f30cd21b6

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/stock_move.py [hunk 1]
```diff
@@ -1155,6 +1160,29 @@ def unlink(self):
         self.mapped('move_line_ids').unlink()
         return super(StockMove, self).unlink()
 
+    def split_mts(self):
+        total_org = sum(qty for qty in self.move_dest_ids.mapped('move_orig_ids.product_uom_qty'))
```
Comment (fmdl) [inline addons/stock/models/stock_move.py]: it is multi unit proof (what happend if move_orig_ids are diffrent unit) ?

File: addons/stock/models/stock_move.py [hunk 2]
```diff
@@ -1155,6 +1160,29 @@ def unlink(self):
...
+        total_dest = sum(qty for qty in self.move_dest_ids.mapped('product_uom_qty'))
```
Comment (fmdl) [inline addons/stock/models/stock_move.py]: don't need for loop
total_dest = sum(self.move_dest_ids.mapped('product_uom_qty')

File: addons/stock/models/stock_move.py [hunk 3]
```diff
@@ -1155,6 +1160,29 @@ def unlink(self):
...
+        if total_org < total_dest:
+            for dest_move in self.move_dest_ids:
+                if dest_move.procure_method == 'make_to_order':
+                    qty_split = dest_move.product_uom_qty - dest_move.reserved_availability
+                    if qty_split == dest_move.product_uom_qty:
```
Comment (fmdl) [inline addons/stock/models/stock_move.py]: maybe use `float_compare`

File: addons/stock/models/stock_move.py [hunk 4]
```diff
@@ -1155,6 +1160,29 @@ def unlink(self):
...
+                        dest_move.write({
+                            'move_orig_ids': [(5, 0, 0)],
+                            'procure_method': 'make_to_stock',
+                        })
+                        dest_move._action_assign()
+                    elif qty_split > 0:
```
Comment (fmdl) [inline addons/stock/models/stock_move.py]: maybe use `float_compare`

File: addons/stock/models/stock_move.py [hunk 5]
```diff
@@ -1155,6 +1160,29 @@ def unlink(self):
         self.mapped('move_line_ids').unlink()
         return super(StockMove, self).unlink()
 
+    def split_mts(self):
```
Comment (fmdl) [inline addons/stock/models/stock_move.py]: get a private methode (with underscore)  `def _split_mts(self):`
and add `self.ensure_one()`

**Review by fmdl — CHANGES_REQUESTED** [review]

**Comment by C3POdoo** [comment]
Dear @kirti-c,

Thank you for your report but we are closing it due to inactivity.
We apology if we could not look at your request in time.
If your report still makes sense, don't hesitate to reopen a new one. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**